### PR TITLE
added 'public' to many methods so I can extend the Fetcher class outside of the Haneke module

### PR DESCRIPTION
**Pull request by szehnder**

### CODE REVIEW

**Comment by hpique** [comment]
@szehnder Thanks for catching this! I'm not keen on making public most of these, but the Fetcher class should definitely be. If you change this I can merge this PR. Otherwise I'll close and add the public myself later.

File: Haneke/Fetcher.swift [hunk 1]
```diff
@@ -13,29 +13,29 @@ public class Fetcher<T : DataConvertible> {
...
     
-    func fetch(failure fail : ((NSError?) -> ()), success succeed : (T.Result) -> ()) {}
+    public func fetch(failure fail : ((NSError?) -> ()), success succeed : (T.Result) -> ()) {}
     
-    func cancelFetch() {}
+    public func cancelFetch() {}
 }
 
-class SimpleFetcher<T : DataConvertible> : Fetcher<T> {
+public class SimpleFetcher<T : DataConvertible> : Fetcher<T> {
```
Comment (hpique) [inline Haneke/Fetcher.swift]: SimpleFetcher doesn't need to be public. It's an internal convenience class.

File: Haneke/Haneke.swift [hunk 2]
```diff
@@ -49,14 +49,14 @@ public struct Shared {
     }
 }
 
-func errorWithCode(code : Int, #description : String) -> NSError {
+public func errorWithCode(code : Int, #description : String) -> NSError {
```
Comment (hpique) [inline Haneke/Haneke.swift]: This should not be public. I don't want API clients to generate Haneke-labeled errors.

File: Haneke/Haneke.swift [hunk 3]
```diff
@@ -49,14 +49,14 @@ public struct Shared {
...
     let userInfo = [NSLocalizedDescriptionKey: description]
     return NSError(domain: HanekeGlobals.Domain, code: code, userInfo: userInfo)
 }
 
-struct Log {
+public struct Log {
```
Comment (hpique) [inline Haneke/Haneke.swift]: Same reasoning as above.

File: Haneke/NSHTTPURLResponse+Haneke.swift [hunk 4]
```diff
@@ -8,9 +8,9 @@
 
 import Foundation
 
-extension NSHTTPURLResponse {
+public extension NSHTTPURLResponse {
```
Comment (hpique) [inline Haneke/NSHTTPURLResponse+Haneke.swift]: I don't want to make this "official" API as it might change at any time.

**Comment by szehnder** [comment]
Awesome, yeah I was moving fast.  I sincerely appreciate your careful response and I'll make the modifications you've pointed out.  Shouldn't take me but a few mins.

**Pull request closed without merging**